internal/controller/nopresource/nopresource.go

```diff
@@ -0,0 +1,220 @@
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	errNotNopResource = "managed resource is not a NopResource custom resource"
+)
+
+// Setup adds a controller that reconciles NopResource managed resources.
+func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha1.NopResourceGroupKind)
+
+	r := managed.NewReconciler(mgr,
+		resource.ManagedKind(v1alpha1.NopResourceGroupVersionKind),
+		managed.WithExternalConnecter(&connector{
+			kube: mgr.GetClient(),
+		}),
+		managed.WithInitializers(managed.NewNameAsExternalName(mgr.GetClient())),
+		managed.WithPollInterval(1*time.Second),
+		managed.WithLogger(l.WithValues("controller", name)),
+		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.NopResource{}).
+		Complete(r)
+}
+
+// A connector is expected to produce an ExternalClient when its Connect method
+// is called.
+type connector struct {
+	kube client.Client
+}
+
+// Connect typically produces an ExternalClient by:
+// 1. Tracking that the managed resource is using a ProviderConfig.
+// 2. Getting the managed resource's ProviderConfig.
+// 3. Getting the credentials specified by the ProviderConfig.
+// 4. Using the credentials to form a client.
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	_, ok := mg.(*v1alpha1.NopResource)
+	if !ok {
+		return nil, errors.New(errNotNopResource)
+	}
+
+	// fmt.Printf(string(cr.GetCondition(xpv1.TypeReady).Status) + "\n\n")
+
+	return &external{}, nil
+}
+
+// An ExternalClient observes, then either creates, updates, or deletes an
+// external resource to ensure it reflects the managed resource's desired state.
+type external struct {
+	// A 'client' used to connect to the external resource API. In practice this
+	// would be something like an AWS SDK client.
+	service interface{}
+}
+
+func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.NopResource)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotNopResource)
+	}
+	startTime := cr.CreationTimestamp
+
+	// If object was deleted, return it does not exist so that managed reconciler removes finalizer
+	if meta.WasDeleted(mg) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+
+	ci := reconcileLogic(cr.Spec.ForProvider.ConditionAfter, time.Since(startTime.Time))
+
+	for _, l := range ci {
+		// fmt.Printf("Calling update on index %d\n", l)
+
+		x := xpv1.Condition{
+			Type:               xpv1.ConditionType(cr.Spec.ForProvider.ConditionAfter[l].ConditionType),
+			Status:             v1.ConditionStatus(cr.Spec.ForProvider.ConditionAfter[l].ConditionStatus),
+			LastTransitionTime: metav1.Now(),
+			Reason:             xpv1.ReasonAvailable,
```

Comment:
As @jbw976 pointed out, 
> this Reason is not always going to be accurate depending on what the condition actually is - Ready (true/false) and Synced (true/false) all have different reasons associated with them: https://github.com/crossplane/crossplane-runtime/blob/master/apis/common/v1/condition.go#L43-L54

Will it be a good way to map Type and Status with a Reason to resolve this?